src/Test/QuickCheck/Test.hs

```diff
@@ -343,30 +397,30 @@ showTestCount st =
             ]
 
 runATest :: State -> Property -> IO Result
-runATest st f =
+runATest st prop =
   do -- CALLBACK before_test
      putTemp (terminal st)
         ( "("
        ++ showTestCount st
        ++ ")"
         )
-     let powerOfTwo n = n .&. (n - 1) == 0
-     let f_or_cov =
-           case coverageConfidence st of
-             Just confidence | (1 + numSuccessTests st) `mod` 100 == 0 && powerOfTwo ((1 + numSuccessTests st) `div` 100) ->
-               addCoverageCheck confidence st f
-             _ -> f
+
      let size = computeSize st
-     MkRose res ts <- protectRose (reduceRose (unProp (unGen (unProperty f_or_cov) rnd1 size)))
+
+     MkRose res ts <- protectRose (reduceRose (unProp (unGen (unProperty prop) rnd1 size)))
      res <- callbackPostTest st res
 
-     let continue break st' | abort res = break (addNewOptions st')
-                            | otherwise = test (addNewOptions st')
+     let continue break st'
+           | abort res = break $ updateState st'
+           | otherwise = test $ updateState st'
+
+         updateState st0 = addNewOptions $ st0{ randomSeed = rnd2 }
 
-         addNewOptions st0 = st0{ maxSuccessTests = fromMaybe (maxSuccessTests st0) (maybeNumTests res)
+         addNewOptions st0 = st0{ maxSuccessTests   = fromMaybe (maxSuccessTests st0) (maybeNumTests res)
                                 , maxDiscardedRatio = fromMaybe (maxDiscardedRatio st0) (maybeDiscardedRatio res)
-                                , numTotMaxShrinks = fromMaybe (numTotMaxShrinks st0) (maybeMaxShrinks res)
-                                , maxTestSize = fromMaybe (maxTestSize st0) (maybeMaxTestSize res)
+                                , numTotMaxShrinks  = fromMaybe (numTotMaxShrinks st0) (maybeMaxShrinks res)
+                                , maxTestSize       = fromMaybe (maxTestSize st0) (maybeMaxTestSize res)
+                                , expected          = expect res
```

Comment:
We now set `expected` also in the discard case, but I don't think it makes any difference..?

Reply:
It shouldn't.